### DISCUSSION TITLE
<Button as="a" /> with React Router

### DISCUSSION THREAD

**Discussion by and-lopes** [twilio-labs/paste#2612] [General] [answered]
![image](https://user-images.githubusercontent.com/75698844/184245196-bf293cc6-3bff-460b-aa0d-3be2d339492a.png)

PN currently utilizes a combination of `Tab` and `Buttons as="a"` to organize links to different Incoming Phone Number pages. This allows us to mask the link within the `Tab` without an underline while abiding by accessibility guidelines.

This implementation triggers a page refresh, however, so I'm looking to fix this. If we use an `Anchor`, we get the underline though. Should we be rethinking this design? I'm leaning towards just using a button with my own implementation: https://github.com/twilio-internal/console/pull/1876

TLDR: Button as="a" lacks flexibility in implementation for custom routing / `Anchor` lacks

**Comment by TheSisb** [comment, marked as answer]
Hi @and-lopes,

Have you seen [this previous discussion](https://github.com/twilio-labs/paste/discussions/1244) where I demonstrate how we can make Tabs re-direct to sub-routes? Let me know if this helps you get going.  As Simon pointed out in that thread, Tabs should not be used as page navigation, they should only switch content. Alas we haven't had the chance to build the component to solve this problem yet. If you're able to contemplate page design, it may be something to keep in mind.

**Reply by and-lopes** [reply]
Hadn't seen that. Thank you!
